### DISCUSSION TITLE
Can someone help me setup RTSP with Wyze V3 / BlueIris?

### DISCUSSION THREAD

**Discussion by mlb22uk** [gtxaspec/wz_mini_hacks#219] [Q&A]
Hi all, im really struggling to get this setup working.

I have copied the contents of SD_ROOT (copied them, not the whole foler titled SD_ROOT (is this correct?) to the formatted SD card, then put that in my wyzze V3 cam.

The camera turns on and works, however I cant get RTSP working in blueiris, just says failed to connect.  The only thing ive changed from the config file is RTSP admin, login and set high quality to enabled.

What am I doing wrong?

Thanks

**Comment by endertable** [comment]
I reckon the first step to this, since you didn't mention it specifically, is whether you have an RTSP feed running on the cam.  
As a quick you can run:  

netstat -tnl

to see if you have your RTSP ports listening but verifying with VLC on iPhone or PC that you get video is ultimately the best verification.

**Reply by Bob-AOL** [reply]
I encountered the same issue yesterday and was graciously (and surprisingly) supported by the host who hooked me right up. The following is essentially the end result of the exchange after eliminating some of the more common potential issues with a fresh flash up.

You will obviously need to set up the RTSP params in the config file for starters. The camera firmware is irrelevant as long as the camera has been initialized and is accessible on your wifi network. Keep everything simple. 

The key is to config Blue Iris to use rtsp://x.x.x.x:554/unicast as the primary url for starters just to get things going. Activating the sub-stream and having both active will require videosource1 and videosource2 respectively instead of unicast -- I think. Haven't actually tried yet. Someone else will surely jump in with clarification. I also used 2048 and 1024 as bitrates (vs. kB/s) and 2 VBR as the enc setting. Tested on both VLC and BI with instant success.

I'll note that I had yet to get ssh set up or explore the ethernet setup prior to getting RTSP working. They are not prerequisites for doing so. All conf file editing was done with Notepad via PC laptop card slot directly on the SD card since I didn't have SSH going yet. (You can also enable the web server which provides a nice little RTSP config template for doing same).

**Reply by mlb22uk** [reply]
Hey! Thanks alot for this, finally got it working!  Can you elobarote on the "activating the sub stream and having both active" what does this mean?  Are you referrering to multiple camera setup?  I have 5 cameras I need to set up on blueiris.

**Comment by Bob-AOL** [comment]
I'm referring to what is usually called the "main" video stream which will typically be the highest resolution stream of a camera and the one usually chosen for recording -- the "HI-RES" setting in the RTSP config. 

The sub-stream is a secondary low resolution stream usually used for live monitoring on multi-view monitors where the images are small and the lack of resolution isn't drastically noticeable. More importantly, using sub-streams dramatically reduce the load on a CPU which in turn allows the use of more cameras on a given surveillance system with only 'x' amount of processing horsepower -- particularly a Blue Iris system. This would be the "LO-RES" setup of the RTSP configuration.

The v3 cameras don't come stock with a configurable sub-stream and are basically just an arbitrary variable bitrate device. The hack offers the ability to define the sub-stream if desired. It's not a necessity, but rather a feature common to most IP cameras.

The following is lifted from the readme file and where the respective settings would be applied in the actual conf file. As mentioned, there is also a handy template via the web server that provides the same configuration capabilities.

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"
RTSP_HI_RES_FPS="15"

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"
RTSP_LOW_RES_MAX_BITRATE=""
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""
RTSP_LOW_RES_FPS=""
```
the singular stream will be located at ```rtsp://login:password@IP_ADDRESS:8554/unicast```
multiple streams are located at ```rtsp://login:password@IP_ADDRESS:8554/video1_unicast``` and ```rtsp://login:password@IP_ADDRESS:8554/video2_unicast```
